### PR TITLE
Open valid url under cursor

### PR DESCRIPTION
**Pull request by anijanyan**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/editor.js [hunk 1]
```diff
@@ -1729,6 +1729,38 @@ Editor.$uid = 0;
         }
     };
 
+    /**
+     * Finds link at defined {row} and {column}
+     * @returns {String}
+     **/
+    this.findLinkAt = function (row,column) {
```
Comment (andrewnester) [inline src/editor.js:1736]: can we add a test for this function?

File: src/editor.js [hunk 2]
```diff
@@ -1729,6 +1729,38 @@ Editor.$uid = 0;
...
+        var line = this.session.getLine(row);
+        var wordParts = line.split(/((?:https?|ftp):\/\/[\S]+)/);
+        var delta = column;
```
Comment (andrewnester) [inline src/editor.js:1739]: can we rename `delta` to `columnPosition` instead to reflect the intent?

File: src/editor.js [hunk 3]
```diff
@@ -1729,6 +1729,38 @@ Editor.$uid = 0;
...
+        if (delta < 0) delta = 0;
+        var curLength = 0, itLength = 0, match;
+        wordParts.forEach(function (item) {
+            itLength = curLength + item.length;
```
Comment (andrewnester) [inline src/editor.js:1743]: can we rename curLength to previousPosition and itLength to currentPosition?

File: src/editor.js [hunk 4]
```diff
@@ -1729,6 +1729,38 @@ Editor.$uid = 0;
...
+            if (delta >= curLength && delta <= itLength) {
+                if (item.match(/((?:https?|ftp):\/\/[\S]+)/)) {
+                    match = item.replace(/[\s:.,'";}\]]+$/, "");
```
Comment (andrewnester) [inline src/editor.js:1746]: should we add break here so we don't look at the rest of the line? it can help with large lines

**Review by andrewnester — CHANGES_REQUESTED** [review]

**Comment by anijanyan** [comment]
@andrewnester Thank you for review! All should be fixed now.

**Review by andrewnester — APPROVED** [review]

**Review by andrewnester — CHANGES_REQUESTED** [review]
Build fails with

> /home/runner/work/ace/ace/src/editor_commands_test.js
Warning:     8:5  warning  'ace' is assigned a value but never used          no-unused-vars
Warning:     9:5  warning  'EditSession' is assigned a value but never used  no-unused-vars
Error:   546:6  error    Unexpected trailing comma                         comma-dangle

> ✖ 3 problems (1 error, 2 warnings)
  1 error and 0 warnings potentially fixable with the `--fix` option.

**Comment by anijanyan** [comment]
> Build fails with
> 
> > /home/runner/work/ace/ace/src/editor_commands_test.js
> > Warning:     8:5  warning  'ace' is assigned a value but never used          no-unused-vars
> > Warning:     9:5  warning  'EditSession' is assigned a value but never used  no-unused-vars
> > Error:   546:6  error    Unexpected trailing comma                         comma-dangle
> 
> > ✖ 3 problems (1 error, 2 warnings)
> > 1 error and 0 warnings potentially fixable with the `--fix` option.

Oh, sorry, haven't noticed this. Fixed now.

**Pull request merged**